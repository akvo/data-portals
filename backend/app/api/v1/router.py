import os
from csv import DictReader
from decimal import Decimal
from typing import Any

from fastapi import APIRouter
from fastapi.responses import FileResponse

router = APIRouter()


@router.get("/functionality-rate-by-region")
async def get_functionality_rate_by_region() -> Any:
    filename = os.path.realpath(
        os.path.join(
            os.getcwd(),
            os.path.dirname(__file__),
            "../../../datasets/Taux de fonctionalitÃ© par region - Blad1.csv",
        )
    )

    with open(filename, "r") as csvfile:
        reader = DictReader(csvfile)
        return [
            {
                "Region": row["REGION"],
                "PMH": Decimal(row["PMH"].rstrip("%").replace(",", ".")),
                "Puits": Decimal(row["Puits "].rstrip("%").replace(",", ".")),
                "SAEP": Decimal(row["SAEP"].rstrip("%").replace(",", ".")),
            }
            for row in reader
        ]


@router.get("/site-points.geojson")
def get_site_points_raw() -> Any:
    filename = os.path.realpath(
        os.path.join(
            os.getcwd(),
            os.path.dirname(__file__),
            "../../../datasets/site_points_deau_mali.geojson",
        )
    )
    return FileResponse(filename)
