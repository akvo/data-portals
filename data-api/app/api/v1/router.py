from fastapi import APIRouter

from app.api.v1 import mali

router = APIRouter()

router.include_router(mali.router, prefix="/mali")
