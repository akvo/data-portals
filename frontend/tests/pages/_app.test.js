import React from 'react'
import CustomApp from '../../pages/_app'
import { renderMockRouter } from '../mocks/router'

const Example = () => {
  return <div>Example</div>
}

describe('CustomApp layout', () => {
  it('matches snapshot', () => {
    const { asFragment } = renderMockRouter(<CustomApp Component={Example} />, {
      router: { pathname: '/' },
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
