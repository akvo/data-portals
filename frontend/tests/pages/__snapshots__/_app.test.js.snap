// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomApp layout matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="ant-layout"
  >
    <header
      class="ant-layout-header layout-header"
    >
      <div
        class="logo"
      >
        <a
          href="/"
        >
          <img
            src="/logo.svg"
          />
        </a>
      </div>
      <ul
        class="ant-menu ant-menu-dark ant-menu-root ant-menu-horizontal"
        role="menu"
      >
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="display: none;"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-item ant-menu-item-only-child"
          role="menuitem"
        >
          <a
            href="/water"
          >
            Water
          </a>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="display: none;"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-item ant-menu-item-only-child"
          role="menuitem"
        >
          <a
            href="/water-quality"
          >
            Water quality
          </a>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="display: none;"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-item ant-menu-item-only-child"
          role="menuitem"
        >
          <a
            href="/sanitation"
          >
            Sanitation
          </a>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="display: none;"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-item ant-menu-item-only-child"
          role="menuitem"
        >
          <a
            href="/hygiene"
          >
            Hygiene
          </a>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="display: none;"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-item ant-menu-item-only-child"
          role="menuitem"
        >
          <a
            href="/data"
          >
            Data
          </a>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="display: none;"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-item ant-menu-item-only-child"
          role="menuitem"
        >
          <a
            href="/guide"
          >
            Guide
          </a>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="visibility: hidden; position: absolute; display: none;"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
      </ul>
    </header>
    <main
      class="ant-layout-content layout-content"
    >
      <div>
        Example
      </div>
    </main>
    <footer
      class="ant-layout-footer"
    >
      <p>
        @2020
      </p>
    </footer>
  </section>
</DocumentFragment>
`;
